File: frontend/src/app/core/services/review.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { ApiResponse, ReviewDto, SubmitReviewRequest, MentorRatingDto, PageResponse } from '../../shared/models';

@Injectable({ providedIn: 'root' })
export class ReviewService {
  private readonly http = inject(HttpClient);
  private readonly base = `${environment.apiUrl}/review`;

  submitReview(req: SubmitReviewRequest): Observable<ApiResponse<ReviewDto>> {
    return this.http.post<ApiResponse<ReviewDto>>(this.base, req);
  }

  getReview(id: number): Observable<ApiResponse<ReviewDto>> {
    return this.http.get<ApiResponse<ReviewDto>>(`${this.base}/${id}`);
  }

  getMentorReviews(mentorId: number, page: number = 0, size: number = 10): Observable<ApiResponse<PageResponse<ReviewDto>>> {
    return this.http.get<ApiResponse<PageResponse<ReviewDto>>>(`${this.base}/mentors/${mentorId}`, {
      params: { page: page.toString(), size: size.toString() }
    });
  }

  getMentorRating(mentorId: number): Observable<ApiResponse<MentorRatingDto>> {
    return this.http.get<ApiResponse<MentorRatingDto>>(`${this.base}/mentors/${mentorId}/rating`);
  }

  updateReview(id: number, req: SubmitReviewRequest): Observable<ApiResponse<ReviewDto>> {
    return this.http.put<ApiResponse<ReviewDto>>(`${this.base}/${id}`, req);
  }

  deleteReview(id: number): Observable<ApiResponse<void>> {
    return this.http.delete<ApiResponse<void>>(`${this.base}/${id}`);
  }
}

File: backend/review-service/src/main/java/com/skillsync/review/service/ReviewService.java
package com.skillsync.review.service;

import com.skillsync.review.dto.request.SubmitReviewRequestDto;
import com.skillsync.review.dto.response.MentorRatingDto;
import com.skillsync.review.dto.response.PageResponse;
import com.skillsync.review.dto.response.ReviewResponseDto;
import java.util.List;

public interface ReviewService {
    ReviewResponseDto submitReview(Long learnerId, SubmitReviewRequestDto request);
    ReviewResponseDto getReview(Long reviewId);
    PageResponse<ReviewResponseDto> getMentorReviews(Long mentorId, int page, int size);
    PageResponse<ReviewResponseDto> getLearnerReviews(Long learnerId, int page, int size);
    MentorRatingDto getMentorRating(Long mentorId);
    ReviewResponseDto updateReview(Long reviewId, Long learnerId, SubmitReviewRequestDto request);
    void deleteReview(Long reviewId, Long learnerId);
}
